components/DonatedItemsList.tsx

```diff
@@ -60,16 +75,33 @@ const DonatedItemsList: React.FC<DisplayDonatedItemsProps> = ({ userId }) => {
                   postable={item.postable}
                   itemId={item.id}
                 />
-                <div className='flex flex-row'>
+                <div className='flex flex-row gap-2'>
                   <Link href={`/edit-item/${item.id}`}>
-                    <p className='button button-rounded mx-2 my-2'>Edit item</p>
+                    <p className='button button-rounded my-2'>Edit item</p>
                   </Link>
                   <Modal
                     name='Delete Item'
                     itemId={item.id}
                     message='By pressing Confirm you will delete this item'
                     onDeleteSuccess={() => handleDeleteSuccess(item.id!)}
                   />
+                  {item.is_reserved ? (
+                    <UnreserveButton
+                      itemId={item.id}
+                      onReserveStatusChange={() =>
+                        onReserveStatusChange(item.id!)
+                      }
+                    />
+                  ) : (
+                    <ReserveForUserModal
+                      name='Mark as Reserved'
+                      itemId={item.id!}
+                      onReserveStatusChange={() =>
+                        onReserveStatusChange(item.id!)
+                      }
+                      requestedToReserveUserIds={item.requestedToReserve ?? []}
```

Comment:
I love `nullish coalescing operators` but this is not the place for one.

We don't want to lie to our components - if there are no requests to reserve the item then we should be handling already somewhere the return value. This here could hide a bug for us later by disguising a `null` or `undefined` from our sight.

When we call our data let's check the return value and if it's empty we pass `[ ]` out of our util, but the component shouldn't be making any decisions about the value of the prop it's being passed